File: rdds/2.customer.scala
object customer extends App{

    val sc = new SparkContext(local[*], "customer")
    val input = sc.textFile("c:\Users\pp255070\Downloads\customerorders.csv")

    val required = input.map( x => ((x.split(",")(0), x.split(",")(2).toFloat)))
    val aggregated = required.reduceByKey( (x,y) => x+y)
    val sorted = aggregated.sortBy( x => x._2)
    val result = sorted.collect       // this is not rdd as it is not distributed others are rdds as they are distributed

}

File: rdds/2.customer.py
from pyspark import SparkContext

sc = SparkContext("local[*]", "customer")
rdd1 = sc.textFile("path")

rdd2 = rdd1.map(lambda x: (x.split(",")[0] , float(x.split(",")[2])) )

rdd3 = rdd2.reduceByKey(lambda x,y : (x+y))

rdd4 = rdd3.sortBy(lambda x : x[1] ,False)

result = rdd4.collect()

for a in result:
    print(a)

